compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt

```diff
@@ -0,0 +1,177 @@
+    ) {
+        PlainTooltipExample()
+        RichTooltipExample()
+        AdvancedRichTooltipExample()
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+// [START android_compose_components_plaintooltipexample]
+@Composable
+fun PlainTooltipExample(
+    modifier: Modifier = Modifier,
+    plainTooltipText: String = "Add to favorites"
+) {
+    TooltipBox(
+        modifier = modifier,
+        positionProvider = TooltipDefaults.rememberPlainTooltipPositionProvider(),
+        tooltip = {
+            PlainTooltip { Text(plainTooltipText) }
+        },
+        state = rememberTooltipState()
+    ) {
+        IconButton(onClick = { /* Icon button's click event */ }) {
+            Icon(
+                imageVector = Icons.Filled.Favorite,
+                contentDescription = "Localized Description"
+            )
+        }
+    }
+}
+// [END android_compose_components_plaintooltipexample]
+
+@Preview
+@Composable
+private fun PlainTooltipSamplePreview() {
+    PlainTooltipExample()
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+// [START android_compose_components_richtooltipexample]
+@Composable
+fun RichTooltipExample(
+    modifier: Modifier = Modifier,
+    richTooltipSubheadText: String = "Rich Tooltip",
+    richTooltipText: String = "Rich tooltips supports multiple lines of informational text."
+) {
+    val tooltipState = rememberTooltipState(isPersistent = true)
+
+    TooltipBox(
+        modifier = modifier,
+        positionProvider = TooltipDefaults.rememberRichTooltipPositionProvider(),
+        tooltip = {
+            RichTooltip(
+                title = { Text(richTooltipSubheadText) }
+            ) {
+                Text(richTooltipText)
+            }
+        }, state = tooltipState
+    ) {
+        IconButton(onClick = { /* Icon button's click event */ }) {
+            Icon(
+                imageVector = Icons.Filled.Info,
+                contentDescription = "Localized Description"
+            )
+        }
+    }
+}
+// [END android_compose_components_richtooltipexample]
+
+@Preview
+@Composable
+private fun RichTooltipSamplePreview() {
+    RichTooltipExample()
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+// [START android_compose_components_advancedrichtooltipexample]
+@Composable
+fun AdvancedRichTooltipExample(
+    modifier: Modifier = Modifier,
+    richTooltipSubheadText: String = "Custom Rich Tooltip",
+    richTooltipText: String = "Rich tooltips supports multiple lines of informational text.",
+    richTooltipActionText: String = "Dismiss"
+) {
+    val tooltipState = rememberTooltipState(isPersistent = true)
+
+    TooltipBox(
+        modifier = modifier,
+        positionProvider = TooltipDefaults.rememberRichTooltipPositionProvider(),
+        tooltip = {
+            RichTooltip(
+                title = { Text(richTooltipSubheadText) },
+                action = {
+                    Row {
+                        TextButton(onClick = { /* Do something... */ }) {
```

Comment:
I think we can make the dismiss action work, that would be nice

Reply:
Done.